Client/Coordinators/Router/DefaultRouter.swift

```diff
@@ -0,0 +1,99 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import UIKit
+
+class DefaultRouter: NSObject, Router {
+    private var completions: [UIViewController: () -> Void]
+
+    var rootViewController: UIViewController? {
+        return navigationController.viewControllers.first
+    }
+
+    let navigationController: NavigationController
+
+    init(navigationController: NavigationController) {
+        self.navigationController = navigationController
+        self.completions = [:]
+        super.init()
+        self.navigationController.delegate = self
+    }
+
+    func present(_ viewController: UIViewController, animated: Bool = true, completion: (() -> Void)? = nil) {
+        if let completion = completion {
+            completions[viewController] = completion
+        }
+
+        viewController.presentationController?.delegate = self
+        navigationController.present(viewController, animated: animated, completion: nil)
+    }
+
+    func dismiss(animated: Bool = true, completion: (() -> Void)? = nil) {
+        navigationController.dismiss(animated: animated, completion: completion)
+    }
+
+    func push(_ viewController: UIViewController, animated: Bool = true, completion: (() -> Void)? = nil) {
+        // Avoid pushing UINavigationController onto stack
+        guard viewController is UINavigationController == false else {
```

Comment:
After reading the PR a bit more I think I understand. If you want to push a new UINavigationController it needs to be tied to a Router so you can't just push it like any other view controller. Otherwise we won't be able to monitor the dismissals or backs etc. At least I think that might be it 🤔

Reply:
I think it's that navigation controller should be presented and not pushed. I tried pushing a navigation controller and get this error.
<img width="563" alt="Screen Shot 2023-03-29 at 10 13 37 AM" src="https://user-images.githubusercontent.com/11338480/228566536-f673f5c8-7b43-47b0-863a-db11ad37eec3.png">

Reply:
Oh yeah 🙈

Reply:
It's probably better to either remove the guard or add an assert in the return case. Otherwise it will fail silently if a dev mistakenly tries and they might not realize what has gone wrong, whereas at least if you let it try to push or add an assert yourself then at least they will get an error they can use.

Reply:
Removed the guard 👍